build_cpp.sh

```diff
@@ -75,18 +80,27 @@ if [ "$PREFIX_PATH" = "" ]; then
   print_help
   exit 1
 fi
+
+if [ "$ROOT_PATH" = "" ]; then
+  print_help
+  exit 1
+fi
+
 : ${RBA_TOOLCHAIN:=$ANDROID_NDK/build/cmake/android.toolchain.cmake}
 
 # get the prefix path
 prefix=$(cd $PREFIX_PATH && pwd)
-TARGET_PATH=$prefix/target
+TARGET_PATH=$(cd $ROOT_PATH && pwd)
```

Comment:
Why not call the argument `TARGET_PATH` directly?

I see that there are 3 paths here:
- Root path or `CMAKE_FIND_ROOT_PATH` (let's say, output)
- Target path or install space  (output/target)
- Catkin workspace (output/catkin_ws for ROS packages)

If we admit that the install space will be under `output` always, you only need to send 2 of them to the script; I prefer sending `prefix` as before instead of `target` and then the user workspace to be compiled.

Perhaps we can also make use of the `--extend` option as well.

Reply:
Yes, naming is quite confusing here.
I will modify it in the following way:
- One argument is output/ and not output/target dir.
- The other is the catkin_ws to be build (could be output/catkin_ws or example_ws)

Maybe the first one could be called -p --prefix and the other -ws --work-space.